src/arrayMethodSort.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompareFun = (a, b) => {
+      const strA = String(a);
+      const strB = String(b);
+
+      return strA < strB ? -1 : strA > strB ? 1 : 0;
+    };
+
+    const compare =
+      typeof compareFunction === 'function'
+        ? compareFunction
+        : defaultCompareFun;
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          const temp = this[j];
+
+          this[j] = this[j + 1];
+          this[j + 1] = temp;
+        }
+      }
+    }
```

Comment:
You've implemented a bubble sort algorithm, which has a time complexity of O(n^2). This is not efficient for larger arrays. While it's a valid implementation for learning purposes, in a production environment you would want to use a more efficient sorting algorithm like quicksort or mergesort.